### PR TITLE
VC-4233: Create db schema assets

### PR DESCRIPTION
**Pull request by vovan1983**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: database/schema/13-assets.sql [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+CREATE TABLE IF NOT EXISTS vipcoin_chain_assets_create (
+   creator     	    TEXT        NOT NULL,                   -- create assets creator
+   name			    TEXT        NOT NULL,                   -- create assets extra name
+   issuer         	TEXT        NOT NULL,					-- create assets issuer
+   policies    	    JSONB,									-- create assets policies
+   state          	INT,									-- create assets state
+   properties  	    JSONB,									-- create assets properties
+   extras         	JSONB									-- create assets extras
+);
```
Comment (IamStubborN) [inline database/schema/13-assets.sql]: format please

File: database/schema/13-assets.sql [hunk 2]
```diff
@@ -0,0 +1,47 @@
+-- +migrate Up
+CREATE TABLE IF NOT EXISTS vipcoin_chain_assets_assets (
+    issuer          TEXT        NOT NULL,					-- assets issuer
+    name            TEXT        NOT NULL,					-- assets name
+    policies        JSONB,									-- assets policies
```
Comment (IamStubborN) [inline database/schema/13-assets.sql]: INT[]

File: database/schema/13-assets.sql [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    name		    TEXT        NOT NULL,                  -- set extra name
+    extras          JSONB                                  -- set extra extras
+);
+
+CREATE TABLE IF NOT EXISTS vipcoin_chain_assets_manage (
+   creator     	    TEXT        NOT NULL,                  -- manage assets creator
+   name		    	TEXT        NOT NULL,                  -- manage assets extra name
+   policies    	    JSONB,								   -- manage assets policies
+   state       	    INT,								   -- manage assets state
+   properties  	    JSONB,								   -- manage assets properties
```
Comment (IamStubborN) [inline database/schema/13-assets.sql]: INT[]

File: database/schema/13-assets.sql [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    burned          BIGINT,									-- assets burned
+    withdrawn       BIGINT,									-- assets withdrawn
+    in_circulation  BIGINT,									-- assets in_circulation
+    properties      JSONB,									-- assets properties
+    extras          JSONB									-- assets extras
+);
+
+CREATE TABLE IF NOT EXISTS vipcoin_chain_assets_set_extra (
+    creator         TEXT        NOT NULL,                  -- set extra creator
+    name		    TEXT        NOT NULL,                  -- set extra name
```
Comment (alex-bodnar) [inline database/schema/13-assets.sql]: Замени табы на пробелы.

File: database/schema/13-assets.sql [hunk 5]
```diff
@@ -0,0 +1,47 @@
+-- +migrate Up
+CREATE TABLE IF NOT EXISTS vipcoin_chain_assets_assets (
+    issuer          TEXT        NOT NULL,					-- assets issuer
```
Comment (alex-bodnar) [inline database/schema/13-assets.sql]: PRIMARY KEY ?

File: database/schema/13-assets.sql [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+   burned         	BIGINT,								   -- manage assets burned
+   withdrawn      	BIGINT,								   -- manage assets withdrawn
+   in_circulation 	BIGINT								   -- manage assets in_circulation
+);
+
+CREATE TABLE IF NOT EXISTS vipcoin_chain_assets_create (
+   creator     	    TEXT        NOT NULL,                   -- create assets creator
+   name			    TEXT        NOT NULL,                   -- create assets extra name
+   issuer         	TEXT        NOT NULL,					-- create assets issuer
+   policies    	    JSONB,									-- create assets policies
```
Comment (IamStubborN) [inline database/schema/13-assets.sql]: INT[]

**Review by IamStubborN — CHANGES_REQUESTED** [review]

File: database/schema/13-assets.sql (hunk 2 above)
Comment (alex-bodnar) [inline database/schema/13-assets.sql]: Зачем JSONB если это масив int32?
Замени на INT[]

File: database/schema/13-assets.sql [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+CREATE TABLE IF NOT EXISTS vipcoin_chain_assets_assets (
+    issuer          TEXT        NOT NULL,					-- assets issuer
+    name            TEXT        NOT NULL,					-- assets name
+    policies        JSONB,									-- assets policies
+    state           INT,									-- assets state
+    issued          BIGINT,									-- assets issued
+    burned          BIGINT,									-- assets burned
+    withdrawn       BIGINT,									-- assets withdrawn
+    in_circulation  BIGINT,									-- assets in_circulation
+    properties      JSONB,									-- assets properties
```
Comment (alex-bodnar) [inline database/schema/13-assets.sql]: Тоже не вижу смысла использовать JSONB, там 2 поля в этой структуре, лучше их добавить

**Review by alex-bodnar — CHANGES_REQUESTED** [review]

File: database/schema/13-assets.sql [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    extras  JSONB          -- set extra extras
+);
+
+CREATE TABLE IF NOT EXISTS vipcoin_chain_assets_manage
+(
+    creator        TEXT NOT NULL, -- manage assets creator
+    name           TEXT NOT NULL, -- manage assets extra name
+    policies       JSONB,         -- manage assets policies
+    state          INT,           -- manage assets state
+    properties     INT[],         -- manage assets properties
```
Comment (alex-bodnar) [inline database/schema/13-assets.sql]: Не INT[]
Либо:
precision   INT
fee_percent INT
Либо
properties     JSONB

File: database/schema/13-assets.sql [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    creator TEXT NOT NULL, -- set extra creator
+    name    TEXT NOT NULL, -- set extra name
+    extras  JSONB          -- set extra extras
+);
+
+CREATE TABLE IF NOT EXISTS vipcoin_chain_assets_manage
+(
+    creator        TEXT NOT NULL, -- manage assets creator
+    name           TEXT NOT NULL, -- manage assets extra name
+    policies       JSONB,         -- manage assets policies
```
Comment (alex-bodnar) [inline database/schema/13-assets.sql]: INT[]

File: database/schema/13-assets.sql (hunk 8 above)
Comment (IamStubborN) [reply database/schema/13-assets.sql]: precision INT
fee_percent INT

File: database/schema/13-assets.sql [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+);
+
+CREATE TABLE IF NOT EXISTS vipcoin_chain_assets_create
+(
+    creator    TEXT NOT NULL, -- create assets creator
+    name       TEXT NOT NULL, -- create assets extra name
+    issuer     TEXT NOT NULL, -- create assets issuer
+    policies   INT[],         -- create assets policies
+    state      INT,           -- create assets state
+    properties JSONB,         -- create assets properties
```
Comment (IamStubborN) [inline database/schema/13-assets.sql]: precision INT
fee_percent INT

**Comment by IamStubborN** [comment]
Нужно сквош коммитов сделать, сейчас их два одинаковых

**Review by alex-bodnar — APPROVED** [review]
LGTM

**Comment by IamStubborN** [comment]
LGTM

**Review by IamStubborN — APPROVED** [review]

**Pull request merged**